applications/adv_networking_bench/Dockerfile

```diff
@@ -17,22 +17,117 @@
 ARG BASE_IMAGE
 ARG GPU_TYPE
 
-FROM ${BASE_IMAGE} as base
-
+FROM ${BASE_IMAGE} AS base
 ARG UBUNTU_VERSION=22.04
 ARG OFED_VERSION=24.07-0.6.1.0
 ARG CACHEBUST=1
-ARG BUILDARCH
-RUN echo "Using build architecture ${BUILDARCH}"
-RUN echo "Using architecture $TARGETPLATFORM"
+ARG TARGETARCH
+RUN echo "Using architecture ${TARGETARCH}"
+
 
-RUN apt update && apt install -y python3-pyelftools ninja-build meson
+RUN apt update && apt install -y python3-pyelftools ninja-build meson libyaml-cpp-dev
 RUN pip3 install scipy loguru attrs
 WORKDIR /opt
 
-RUN    eval `dpkg-architecture` \
-        && wget https://www.mellanox.com/downloads/DOCA/DOCA_v2.7.0/host/doca-host_2.7.0-204000-24.04-ubuntu2204_${BUILDARCH}.deb -O doca-host.deb \
-        && ls -lh && apt install ./doca-host.deb \
-        && apt update \
-        && apt install -y doca-all \
-        && apt install -y doca-gpu doca-gpu-dev
+ARG DEBIAN_FRONTEND=noninteractive
+ARG DOCA_REPO_LINK=https://linux.mellanox.com/public/repo/doca/2.8.0/ubuntu22.04/x86_64
+
+RUN if [ "${TARGETARCH}" = "amd64" ]; then \
+        DOCA_REPO_LINK=https://linux.mellanox.com/public/repo/doca/2.8.0/ubuntu22.04/x86_64; \
+    elif [ "$TARGETARCH" = "arm64" ]; then \
+        DOCA_REPO_LINK=https://linux.mellanox.com/public/repo/doca/2.8.0/ubuntu22.04/arm64; \
+    else \
+        echo "Unknown architecture: $TARGETARCH"; \
+        exit 1; \
+    fi \
+    && echo "Using DOCA_REPO_LINK=${DOCA_REPO_LINK}" \
+    && apt install -y --no-install-recommends wget software-properties-common gpg-agent \
+    && wget -qO - ${DOCA_REPO_LINK}/GPG-KEY-Mellanox.pub | apt-key add - \
+    && add-apt-repository "deb [trusted=yes] ${DOCA_REPO_LINK} ./" \
+    && apt update -y \
+    && apt install -y --no-install-recommends doca-sdk-aes-gcm doca-sdk-apsh doca-sdk-argp doca-sdk-comch doca-sdk-comm-channel doca-sdk-common doca-sdk-compress doca-sdk-devemu doca-sdk-dma doca-sdk-dpa doca-sdk-dpdk-bridge doca-sdk-erasure-coding doca-sdk-eth doca-sdk-flow doca-sdk-pcc doca-sdk-rdma doca-sdk-sha doca-sdk-telemetry-exporter doca-sdk-urom doca-apsh-config doca-bench doca-caps doca-comm-channel-admin doca-pcc-counters doca-sha-offload-engine doca-socket-relay doca-all doca-sdk-gpunetio libdoca-sdk-gpunetio-dev rdma-core flexio libyara8
```

Comment:
Not familiar, are DOCA packages here prerequisites for each of the following stages (`doca`, `rivermax`, `dpdk`)? Or should these steps be moved to the `doca` stage?